src/Http/RequestExecutor.ts

```diff
@@ -399,14 +423,25 @@ export class RequestExecutor implements IDisposable {
         opts?: IRequestExecutorOptions): RequestExecutor {
         const { authOptions, documentConventions } = opts || {} as IRequestExecutorOptions;
         const executor = new RequestExecutor(database, authOptions, documentConventions);
-        executor._firstTopologyUpdatePromise = executor._firstTopologyUpdate(initialUrls, this.GLOBAL_APPLICATION_IDENTIFIER);
+
+        executor._firstTopologyUpdatePromise = executor._firstTopologyUpdate(initialUrls, RequestExecutor.getGlobalApplicationIdentifier());
 
         // this is just to get rid of unhandled rejection, we're handling it later on
         executor._firstTopologyUpdatePromise.catch(TypeUtil.NOOP);
 
         return executor;
     }
 
+    private static getGlobalApplicationIdentifier() {
+        // due to cloudflare constraints we can't init GLOBAL_APPLICATION_IDENTIFIER in static
+
+        if (!this.GLOBAL_APPLICATION_IDENTIFIER) {
+            this.GLOBAL_APPLICATION_IDENTIFIER = uuidv4();
```

Comment:
Do we need `.toString()` here? It used to be there, and the type changed?
That would be we send `[Object object]` ?

Reply:
uuid4 seems to have toString():
<img width="329" alt="image" src="https://user-images.githubusercontent.com/1250431/226581429-83fe741a-eb64-461e-ac94-43dbc520e5ca.png">